SubsoniciousKit/SubsoniciousKit/Network/Response/SubsonicResponse.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  SubsonicResponse.swift
+//  SubsoniciousKit
+//
+//  Created by Bilal on 05/07/2020.
+//  Copyright © 2020 Bilal Benlarbi. All rights reserved.
+//
+
+import Foundation
+
+public struct SubsonicResponse: Decodable {
+    public let status: Status
+
+    enum CodingKeys: String, CodingKey {
+        case responseContainer = "subsonic-response"
+    }
+
+    enum ResponseCodingKeys: String, CodingKey {
+        case status
+        case error
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        let responseContainer = try container.nestedContainer(keyedBy: ResponseCodingKeys.self, forKey: .responseContainer)
+        status = try responseContainer.decode(Status.self, forKey: .status)
+
+        if let error = try responseContainer.decodeIfPresent(Error.self, forKey: .error) {
+            throw DecodingError.Subsonic.error(error)
+        }
+    }
+}
+
+extension DecodingError {
+    public enum Subsonic: Error {
+        case error(SubsonicResponse.Error)
+    }
+}
+
+extension SubsonicResponse {
+    public enum Status: Decodable {
```

Comment:
you could use enum Status: String, Decodable to automatically conform.

In addition, why do you want to throw you error type instead of default DecodingError for that case ?

Reply:
Good catch 👏